lib/govobject/types/proposal.js

```diff
@@ -95,22 +95,25 @@ Proposal.prototype.fromObject = function fromObject(arg) {
 };
 
 Proposal.prototype.fromBufferReader = function(reader) {
-    $.checkArgument(!reader.finished(), 'No data received');
+  $.checkArgument(!reader.finished(), 'No data received');
 
-    var dataHex = reader.read(reader.buf.length);
-    var object = JSON.parse(dataHex.toString('utf8'));
+  var dataHex = reader.read(reader.buf.length);
+  var object = JSON.parse(dataHex.toString('utf8'));
 
+  if(object.constructor.name === 'Array'){
     this.type = object[0][0];
```

Comment:
Looks good overall, only thing I would change is not getting `this.type` from here, but instead base it on the value in `object[0][1]["type"]`, which is an integer/enum and from where DashCore (and soon Sentinel) takes this info:

https://github.com/dashpay/dash/blob/ef9a9f2d6716f0945a97ebec20771cc3d7907c02/src/governance-object.cpp#L370
https://github.com/dashpay/sentinel/commit/8d9b3c45d579b7b8990aa3a8264fc0f871fa794d

Reply:
Note that this int would then need to lookup the type string in a new hard-coded lookup table:

```
1: "proposal"
2: "trigger" // or "superblock"
```

Reply:
We already do it several lines below, this type nb is also used by the Gobject controller, but the handling of trigger is to be improved (soon)>

Reply:
Ah, ok, see it now. So then, this line `     this.type = object[0][0];` is literally not used at all?

If that's the case we might as well remove it, right?

Reply:
That's how I understand it. I want to add triggers support so I will look at this method and if needed will remove it in another PR

Reply:
Why not to use `Array.isArray`?

Reply:
It could be to, I use the direct constructor.name as I suppose no one would ever try to mutate `object`. Therefore a direct text comparison (is object.constructor.name equal to Array ?) is way more quicker than evaluate the function isArray (which will do a Object.prototype.toString.call(object)and compare it with string `[object Array]`).

Reply:
I don't think it actually makes sense, until your goal is to achive 1 million checks per second :) But if you think that it is better this way, then ok

Reply:
The `it could be too` was meaning quite the opposite, I don't really have too much of any emotionnal affection on this line, I'm good with all ways to ensure our var (like _.has), I just use it more by reflex/habits. 
I'm totally encline to modify it, if you feel that this can improve in any manner (readability, ensuring not counting on a mutable value...) :)